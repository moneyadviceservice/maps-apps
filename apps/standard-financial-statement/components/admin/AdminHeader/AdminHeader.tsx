import Image from 'next/image';

import { twMerge } from 'tailwind-merge';
import { AdminUser } from 'types/admin/base';
import { v4 as uuid } from 'uuid';

import { Icon, IconType, Link } from '@maps-react/common/index';
import { Container } from '@maps-react/core/components/Container';

type Props = {
  assetPath: string;
  logoPath: string;
  user?: AdminUser;
};

export const AdminHeader = ({ assetPath, logoPath, user }: Props) => {
  const skipToContentFocusStyles =
    'focus:not-sr-only focus:bg-yellow-200 focus:z-20 focus:border-b-4 focus:border-b-blue-800 focus:absolute focus:p-2 focus:rounded focus:top-4 focus:left-4 focus:z-50';

  return (
    <header data-testid="admin-header">
      <a className={twMerge('sr-only', skipToContentFocusStyles)} href="#main">
        {'Skip to content'}
      </a>

      <div className="relative bg-blue-600 print:hidden">
        <Container className="flex items-center justify-center  max-w-[1272px]">
          <a
            href={`/admin`}
            aria-label={'SFS Admin'}
            className="max-h-[64px] max-w-[116px] lg:max-w-[186px] lg:max-h-[102px]"
          >
            {logoPath && (
              <Image
                src={`${assetPath}${logoPath}`}
                alt="SFS"
                height={0}
                width={186}
              />
            )}
          </a>
          {user && (
            <div className="items-center hidden ml-auto lg:flex">
              <AccountLinks user={user} />
            </div>
          )}
        </Container>
      </div>
    </header>
  );
};

type AccountLinksProps = {
  user: AdminUser;
};

const AccountLinks = ({ user }: AccountLinksProps) => {
  const accountLinks = [
    {
      linkTo: '/profile',
      item: (
        <div className="flex items-center gap-2">
          <Icon
            type={IconType.PROFILE}
            className="fill-white [&_g]:fill-white"
          />
          <div className="flex flex-col leading-tight">
            <div className="font-bold">{user.name}</div>
            <div>{user.email}</div>
          </div>
        </div>
      ),
      description: null,
      classes: 'no-underline, text-sm',
    },
    {
      linkTo: '/logout',
      item: 'Sign out',
      description: null,
    },
  ];

  return (
    <nav data-testid="admin-nav">
      <ul className="flex items-center space-x-4 ">
        {accountLinks?.map(({ item, linkTo, description, classes }) => {
          return (
            <li key={uuid()} className="flex items-center">
              <Link
                className={twMerge(
                  ['ml-2 flex items-center text-white '],
                  classes,
                )}
                href={`${linkTo}`}
                title={description ?? ''}
              >
                {item}
              </Link>
            </li>
          );
        })}
      </ul>
    </nav>
  );
};
