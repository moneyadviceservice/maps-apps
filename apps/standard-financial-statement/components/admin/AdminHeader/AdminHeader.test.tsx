import { cleanup, render } from '@testing-library/react';

import { AdminHeader } from './AdminHeader';

import '@testing-library/jest-dom';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
    asPath: '/',
  }),
}));

const user = { name: 'Joe Blogs', email: 'email@domain.com' };

describe('Header component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it('renders correctly with user', () => {
    const { getByTestId } = render(
      <AdminHeader
        logoPath={'/logoPath'}
        assetPath={'http://localhost:3000'}
        user={user}
      />,
    );

    const accountNav = getByTestId('admin-nav');
    expect(accountNav).toBeInTheDocument();

    const header = getByTestId('admin-header');
    expect(header).toMatchSnapshot();
  });

  it('renders correctly with no user', () => {
    const { getByTestId, queryByTestId } = render(
      <AdminHeader
        logoPath={'/logoPath'}
        assetPath={'http://localhost:3000'}
      />,
    );

    const adminNav = queryByTestId('admin-nav');
    expect(adminNav).not.toBeInTheDocument();

    const header = getByTestId('admin-header');
    expect(header).toMatchSnapshot();
  });
});
