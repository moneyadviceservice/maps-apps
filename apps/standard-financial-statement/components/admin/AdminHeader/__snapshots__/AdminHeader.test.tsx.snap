// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly with no user 1`] = `
<header
  data-testid="admin-header"
>
  <a
    class="sr-only focus:not-sr-only focus:bg-yellow-200 focus:border-b-4 focus:border-b-blue-800 focus:absolute focus:p-2 focus:rounded focus:top-4 focus:left-4 focus:z-50"
    href="#main"
  >
    Skip to content
  </a>
  <div
    class="relative bg-blue-600 print:hidden"
  >
    <div
      class="container-auto flex items-center justify-center max-w-[1272px]"
    >
      <a
        aria-label="SFS Admin"
        class="max-h-[64px] max-w-[116px] lg:max-w-[186px] lg:max-h-[102px]"
        href="/admin"
      >
        <img
          alt="SFS"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          src="/_next/image?url=http%3A%2F%2Flocalhost%3A3000%2FlogoPath&w=384&q=75"
          srcset="/_next/image?url=http%3A%2F%2Flocalhost%3A3000%2FlogoPath&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2FlogoPath&w=384&q=75 2x"
          style="color: transparent;"
          width="186"
        />
      </a>
    </div>
  </div>
</header>
`;

exports[`Header component renders correctly with user 1`] = `
<header
  data-testid="admin-header"
>
  <a
    class="sr-only focus:not-sr-only focus:bg-yellow-200 focus:border-b-4 focus:border-b-blue-800 focus:absolute focus:p-2 focus:rounded focus:top-4 focus:left-4 focus:z-50"
    href="#main"
  >
    Skip to content
  </a>
  <div
    class="relative bg-blue-600 print:hidden"
  >
    <div
      class="container-auto flex items-center justify-center max-w-[1272px]"
    >
      <a
        aria-label="SFS Admin"
        class="max-h-[64px] max-w-[116px] lg:max-w-[186px] lg:max-h-[102px]"
        href="/admin"
      >
        <img
          alt="SFS"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          src="/_next/image?url=http%3A%2F%2Flocalhost%3A3000%2FlogoPath&w=384&q=75"
          srcset="/_next/image?url=http%3A%2F%2Flocalhost%3A3000%2FlogoPath&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Flocalhost%3A3000%2FlogoPath&w=384&q=75 2x"
          style="color: transparent;"
          width="186"
        />
      </a>
      <div
        class="items-center hidden ml-auto lg:flex"
      >
        <nav
          data-testid="admin-nav"
        >
          <ul
            class="flex items-center space-x-4 "
          >
            <li
              class="flex items-center"
            >
              <a
                class="gap-2 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none ml-2 flex items-center text-white no-underline, text-sm"
                href="/profile"
                title=""
              >
                <div
                  class="flex items-center gap-2"
                >
                  <profile.svg
                    class="fill-white [&_g]:fill-white"
                  />
                  <div
                    class="flex flex-col leading-tight"
                  >
                    <div
                      class="font-bold"
                    >
                      Joe Blogs
                    </div>
                    <div>
                      email@domain.com
                    </div>
                  </div>
                </div>
              </a>
            </li>
            <li
              class="flex items-center"
            >
              <a
                class="gap-2 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none ml-2 flex items-center text-white"
                href="/logout"
                title=""
              >
                Sign out
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</header>
`;
